subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/version/GccVersionDeterminer.java

```diff
@@ -67,47 +71,97 @@ public GccVersionResult getGccMetaData(File gccBinary, List<String> args) {
         }
-        return transform(output, gccBinary);
+        String output = transform.getLeft();
+        String error = transform.getRight();
+        try {
+            return transform(output, error, gccBinary);
+        } catch (BrokenResultException e) {
+            return new BrokenResult(e.getMessage());
+        }
     }
 
     @Override
     public CompilerType getCompilerType() {
         return compilerType;
     }
 
-    private String transform(File gccBinary, List<String> args) {
+    private Pair<String, String> transform(File gccBinary, List<String> args) {
         ExecAction exec = execActionFactory.newExecAction();
         exec.executable(gccBinary.getAbsolutePath());
         exec.setWorkingDir(gccBinary.getParentFile());
         exec.args(args);
         StreamByteBuffer buffer = new StreamByteBuffer();
+        StreamByteBuffer errorBuffer = new StreamByteBuffer();
         exec.setStandardOutput(buffer.getOutputStream());
-        exec.setErrorOutput(NullOutputStream.INSTANCE);
+        exec.setErrorOutput(errorBuffer.getOutputStream());
         exec.setIgnoreExitValue(true);
         ExecResult result = exec.execute();
 
         int exitValue = result.getExitValue();
         if (exitValue == 0) {
-            return buffer.readAsString();
+            return Pair.of(buffer.readAsString(), errorBuffer.readAsString());
         } else {
             return null;
         }
     }
 
-    private GccVersionResult transform(String output, File gccBinary) {
+    private GccVersionResult transform(String output, String error, File gccBinary) {
+        Map<String, String> defines = parseDefines(output, gccBinary);
+        VersionNumber versionNumber = determineVersion(defines, gccBinary);
+        ArchitectureInternal architecture = determineArchitecture(defines);
+        ImmutableList<File> systemIncludes = determineSystemIncludes(error);
+
+        return new DefaultGccVersionResult(versionNumber, architecture, systemIncludes);
+    }
+
+    private ImmutableList<File> determineSystemIncludes(String error) {
+        BufferedReader reader = new BufferedReader(new StringReader(error));
+        String line;
+        ImmutableList.Builder<File> builder = ImmutableList.builder();
+        boolean systemIncludesStarted = false;
+        try {
+            while ((line = reader.readLine()) != null) {
+                if (SYSTEM_INCLUDES_END.equals(line)) {
+                    break;
+                }
+                if (SYSTEM_INCLUDES_START.equals(line)) {
+                    systemIncludesStarted = true;
+                    continue;
+                }
+                if (systemIncludesStarted) {
+                    // Exclude frameworks for CLang - they need to be handled differently
+                    if (compilerType == CompilerType.CLANG && FRAMEWORK_INCLUDE.matcher(line).matches()) {
```

Comment:
Could this be a simpler `line.contains(" (framework directory)")` instead of a regex?